### PR TITLE
Feature/enumerable methods

### PR DESCRIPTION
**Pull request by oscarnava**
For review

### CODE REVIEW

File: my_enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    result
+  end
+
+  # TODO: Accept patterns?
+  def my_all?
+    my_each do |*vals|
+      return false unless yield(*vals)
+    end
+    true
+  end
```
Comment (maelfosso) [inline my_enumerable.rb:32]: Not working
```
> [nil, true, 99].all? 
 => false 
> [nil, true, 99].my_all? 
Traceback (most recent call last):
        7: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        6: from (irb):14
        5: from my_enumerable.rb:28:in `my_all?'
        4: from my_enumerable.rb:6:in `my_each'
        3: from my_enumerable.rb:6:in `each'
        2: from my_enumerable.rb:7:in `block in my_each'
        1: from my_enumerable.rb:29:in `block in my_all?'
LocalJumpError (no block given (yield))
```

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  def my_any?
+    my_each do |*vals|
+      return true if yield(*vals)
+    end
+    false
+  end
```
Comment (maelfosso) [inline my_enumerable.rb:40]: Not working 
```
> [nil, true, 99].any? 
 => true 
> [nil, true, 99].my_any? 
Traceback (most recent call last):
        7: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        6: from (irb):22
        5: from my_enumerable.rb:35:in `my_any?'
        4: from my_enumerable.rb:6:in `my_each'
        3: from my_enumerable.rb:6:in `each'
        2: from my_enumerable.rb:7:in `block in my_each'
        1: from my_enumerable.rb:36:in `block in my_any?'
LocalJumpError (no block given (yield))
```

File: my_enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  def my_none?(&block)
+    !my_any?(&block)
+  end
```
Comment (maelfosso) [inline my_enumerable.rb:44]: Not working 
```
> [nil, false, true].none?  
 => false 
> [nil, false, true].my_none?  
Traceback (most recent call last):
        8: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        7: from (irb):33
        6: from my_enumerable.rb:42:in `my_none?'
        5: from my_enumerable.rb:35:in `my_any?'
        4: from my_enumerable.rb:6:in `my_each'
        3: from my_enumerable.rb:6:in `each'
        2: from my_enumerable.rb:7:in `block in my_each'
        1: from my_enumerable.rb:36:in `block in my_any?'
LocalJumpError (no block given (yield))
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

You write a really nice code. It was a pleasure to read it.
Please, check the comment below and make those changes.

Thanks.

File: my_enumerable.rb (hunk 3 above)
Comment (oscarnava) [reply my_enumerable.rb:44]: Solved problems, My bad :-(

Comment (maelfosso) [reply my_enumerable.rb:44]: :+1:

File: my_enumerable.rb (hunk 2 above)
Comment (maelfosso) [reply my_enumerable.rb:40]: :+1:

File: my_enumerable.rb (hunk 1 above)
Comment (maelfosso) [reply my_enumerable.rb:32]: :+1:

**Review by maelfosso — APPROVED** [review]
# Good job :+1: 

You can merge your project and submit the project completion form.
Good luck with your next project.

**Pull request merged**